controllers/datadogagent/feature/apm/feature.go

```diff
@@ -54,18 +56,39 @@ type apmFeature struct {
 
 	createKubernetesNetworkPolicy bool
 	createCiliumNetworkPolicy     bool
+
+	singleStepInstrumentation *instrumentationConfig
+}
+
+type instrumentationConfig struct {
+	enabled            bool
+	enabledNamespaces  []string
+	disabledNamespaces []string
+	libVersions        map[string]string
 }
 
 // ID returns the ID of the Feature
 func (f *apmFeature) ID() feature.IDType {
 	return feature.APMIDType
 }
 
+func shouldEnableAPM(apmConf *v2alpha1.APMFeatureConfig) bool {
+	if apmConf != nil {
+		return apiutils.BoolValue(apmConf.Enabled)
+	}
+	// SingleStepInstrumentation requires APM Enabled
+	if apmConf.SingleStepInstrumentation != nil &&
```

Comment:
if `apmConf.Enabled == false` function will return `false` and SSI settings don't matter, otherwise it will return `true`. Only time this will be evaluated is when `apmConf == nil`. 

If `apmConf.Enabled == true` and `apmConf.SSI == nil` this still should return true right?

I think having a unit test describe these expectations would be beneficial.

Reply:
Yes that's correct. Essentially, this tests that APM enabled by default AND that it takes precedence over SSI if configured. 
I am adding a test to reflect the `apmConf == nil` behavior, but:
> If apmConf.Enabled == true and apmConf.SSI == nil this still should return true right?
is tested with `v2alpha1 apm enabled, use uds` and `v2alpha1 apm enabled, use uds and host port`.